httpcache.go

```diff
@@ -261,6 +260,10 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 			cachedResp.Status = fmt.Sprintf("%d %s", http.StatusOK, http.StatusText(http.StatusOK))
 			cachedResp.StatusCode = http.StatusOK
 			return cachedResp, nil
+		} else if (err != nil || resp.StatusCode >= 500) &&
+			req.Method == "GET" && keepOnError(req.Header) {
```

Comment:
nit/suggestion: we could call `keepOnError` as `keepCacheOnError` to improve readability. But it's a minor thing so you can ignore it if it does not make much sense.